EKrut-Server/src/ekrut/server/managers/ServerOrderManager.java

```diff
@@ -23,15 +31,51 @@
 public class ServerOrderManager {
 
 	private OrderDAO orderDAO;
+	private UserDAO userDAO;
 	private ServerSessionManager sessionManager;
+	private ServerSalesManager salesManager;
+	private IPaymentProcessor paymentProcessor;
 	
-	public ServerOrderManager(DBController dbCon, ServerSessionManager sessionManager) {
+	public ServerOrderManager(DBController dbCon, ServerSessionManager sessionManager, ServerSalesManager salesManager) {
 		this.orderDAO = new OrderDAO(dbCon);
+		this.userDAO = new UserDAO(dbCon);
 		this.sessionManager = sessionManager;
+		this.salesManager = salesManager;
+		this.paymentProcessor = new StubPaymentProcessor();
+	}
+	
+	private float computeDiscount(Order order, User user) {
+		float discount = 0;
+		LocalDateTime now = LocalDateTime.now();
+		String area = order.getType() == OrderType.PICKUP ? order.getEkrutLocation() : user.getArea();
```

Comment:
The Order.ekrutLocation field describe a 'machine' identifier rather than an 'area' identifier